app/src/main/java/com/tari/android/wallet/ui/activity/onboarding/OnBoardingFlowActivity.kt

```diff
@@ -65,7 +65,7 @@ class OnboardingFlowActivity : BaseActivity(), IntroductionFragment.Listener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        if (sharedPrefsWrapper.getOnboardingStarted()) {
+        if (sharedPrefsWrapper.onboardingStarted) {
```

Comment:
I'm not sure of the logic downstream, but please check that this truth table is followed:

| onboardingStarted | onboardingComplete | Should we delete the wallet? |
|---------------------|---------------------|---------------------|
|  No  | No | Yes |
| No | Yes | No (possible unreachable state) |
| Yes | No | Yes |
| Yes | Yes | No!! |

Reply:
@CjS77 you're right it's misleading this way. There is actually another property named `onboardingCompleted` in shared prefs, which gets set after the onboarding complete at the click of the Continue button after emoji-id creation sequence, and if that one is set SplashActivity will take the user directly to authentication, skipping onboarding - code [here](https://github.com/kukabi/wallet-android/blob/c431f4f966ab5dfae39ec2e5e101795329b4df13/app/src/main/java/com/tari/android/wallet/ui/activity/SplashActivity.kt#L68). So it means, if the user has started being onboarded but exited before completion, delete wallet data (and any test data generated) and re-instantiate it. Note that it does not delete the private key.

Reply:
But it looks very misleading this way. I will fix it 👍

Reply:
Btw, this `if` is unreachable also for state `Yes Yes` in the table.

Reply:
Maybe I can make the line as below to make it more readable:
```kotlin
if (sharedPrefsWrapper.onboardingStarted && !sharedPrefsWrapper.onboardingCompleted)
```

Reply:
That definitely makes it more explicit

Reply:
New commit introduces the field `onboardingWasInterrupted` to `SharedPrefsWrapper`. Please see [here](https://github.com/kukabi/wallet-android/blob/89df8d16bf7bc86817976035e8ed8fba241f13be/app/src/main/java/com/tari/android/wallet/util/SharedPrefsWrapper.kt#L108) and [here](https://github.com/kukabi/wallet-android/blob/89df8d16bf7bc86817976035e8ed8fba241f13be/app/src/main/java/com/tari/android/wallet/ui/activity/onboarding/OnBoardingFlowActivity.kt#L68).